### PR TITLE
add ability to enable/disable plugin per project

### LINKED ISSUE #399: CLion plugin: add opportunity to disable plugin for projects that don't need utbot
**Description**
By default IJ plugins are enabled/disabled for all projects. But user may want to disable utbot for some projects. 

**Expected behavior**
Add a UI component to disable plugin for project. If plugin is disabled it should not send any notifications, not expose any actions to user, and preferably UI components like toolwindows, project panes

### PR DESCRIPTION
**Pull request by vol0n**
closes #399

### CODE REVIEW

**Comment by vol0n** [comment]
Currently when user disables plugin, client don't try to establish connection with server (no heartbeat), current client will be disposed, and almost all actions (except disabling/enabling plugin) will be disabled. In addition widget for setting verbose mode will be hidden. However toolwindows (`UTBot Targets`, `UTBot Logs`) will be visible. It is possible to hide them too, but there's a bug, when tool windows are first set unavailable, their buttons are still visible and only after clicking on the button tool window disappears. So, I decided not to hide them

**Comment by vol0n** [comment]
need to disable settings as well when plugin is disabled

**Comment by vol0n** [comment]
- added a checkbox to disable/enable plugin from settings
- when plugin is disabled/enabled settings ui is hidden/shown like in gif

File: clion-plugin/src/main/kotlin/org/utbot/cpp/clion/plugin/actions/TogglePluginAction.kt [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import org.utbot.cpp.clion.plugin.settings.settings
+
+class TogglePluginAction : ToggleAction() {
+    override fun isSelected(e: AnActionEvent): Boolean {
+        val project = e.project ?: return false
+        updateActionText(e, project)
+        return project.settings.storedSettings.isPluginEnabled
+    }
+
+    override fun setSelected(e: AnActionEvent, state: Boolean) {
```
Comment (alexey-utkin) [inline clion-plugin/src/main/kotlin/org/utbot/cpp/clion/plugin/actions/TogglePluginAction.kt:17]: Better rename the `state` param to the meaning name like `plaginEnabled`.

File: clion-plugin/src/main/kotlin/org/utbot/cpp/clion/plugin/actions/UTBotAction.kt [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+abstract class UTBotBaseAction(
+    text: () -> String? = { null },
+    description: () -> String? = { null },
+    icon: Icon? = null
+) : AnAction(text, description, icon) {
+    constructor(text: String) : this({ text })
+
+    override fun update(e: AnActionEvent) {
+        var isEnabled = false
+        e.project?.let {
```
Comment (alexey-utkin) [inline clion-plugin/src/main/kotlin/org/utbot/cpp/clion/plugin/actions/UTBotAction.kt:17]: It is better to use `if` for such conditions

**Review by alexey-utkin — APPROVED** [review]

**Review by alexey-utkin — APPROVED** [review]

File: clion-plugin/src/main/kotlin/org/utbot/cpp/clion/plugin/settings/UTBotProjectStoredSettings.kt [hunk 3]
```diff
@@ -151,6 +167,15 @@ class UTBotProjectStoredSettings(val project: Project) : PersistentStateComponen
         myState.remotePath = REMOTE_PATH_VALUE_FOR_LOCAL_SCENARIO
     }
 
+    private fun onPluginEnabled() {
+        if (!ApplicationManager.getApplication().isUnitTestMode) {
+            RunOnceUtil.runOnceForProject(project, "Show UTBot quick-start wizard to configure project") {
```
Comment (alexey-utkin) [inline clion-plugin/src/main/kotlin/org/utbot/cpp/clion/plugin/settings/UTBotProjectStoredSettings.kt:172]: Need localization or it is a log message?

**Review by alexey-utkin — APPROVED** [review]

**Pull request merged**